src/App.tsx

```diff
@@ -1,44 +1,56 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
 import classNames from 'classnames';
-import { PostsList } from './components/PostsList';
-import { PostDetails } from './components/PostDetails';
-import { UserSelector } from './components/UserSelector';
-import { Loader } from './components/Loader';
+import { PostsList } from './components/PostList/PostsList';
+import { PostDetails } from './components/PostDetails/PostDetails';
+import { UserSelector } from './components/UserSelector/UserSelector';
+import { Post } from './types/Post';
 
 export const App: React.FC = () => {
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
```

Comment:
It's always better to save id in the state rather than an entire object

Reply:
I saved an entire object to the state, because then I pass it as a props to PostDetails, and I don’t have access to the posts' array in this component to find the needed post by id. Does this mean that I need to move the loading of posts from the server to the app component?

Reply:
nope, you don't have to move fetch logic to App component, pass post id from PostList to App, from App pass post id to PostDetails and make fetch request for one post by id in PostDetails component